transport/http/handler.go

```diff
@@ -82,6 +88,25 @@ func (h handler) callHandler(w http.ResponseWriter, req *http.Request) error {
 
 	ctx := context.Background()
 
+	// Extract opentracing etc baggage from headers
+	// Annotate the inbound context with a trace span
+	carrier := opentracing.HTTPHeadersCarrier(req.Header)
+	spanCtx, _ := h.Tracer.Extract(opentracing.HTTPHeaders, carrier)
+	span := h.Tracer.StartSpan(
+		treq.Procedure,
+		opentracing.StartTime(start),
+		opentracing.Tags{
+			"yarpc.caller":   treq.Caller,
+			"yarpc.service":  treq.Service,
+			"yarpc.encoding": treq.Encoding,
+		},
+		ext.RPCServerOption(spanCtx),
+	)
+	ext.PeerService.Set(span, treq.Caller)
+	// ext.HTTPUrl.Set(span, req.URL.String()) // XXX panics
+	// defer span.Finish() // XXX duplicate span submission
```

Comment:
@yurishkuro Would like to draw your attention to the above lines. The first induces a panic in tests. The second induces a duplicate span submission.